io/disk/disk_info.py

```diff
@@ -113,24 +115,46 @@ def test(self):
         size, UUID and IO sizes etc
         """
         msg = []
-        disk = (self.disk.split("/dev/"))[1]
-        if process.system("ls /dev/disk/by-id -l| grep -i %s" % disk,
-                          ignore_status=True, shell=True, sudo=True) != 0:
-            msg.append("Given disk %s is not in /dev/disk/by-id" % disk)
-        if process.system("ls /dev/disk/by-path -l| grep -i %s" % disk,
+        disk_nodes = []
+        mpath = False
+        if "mapper" in self.disk:
+            mpath = True
+            disk_abs = os.path.basename(os.readlink(self.disk))
+            disk = (self.disk.split("/dev/mapper/"))[1]
+            mwwid = multipath.get_multipath_wwid(disk)
+            disk_nodes = multipath.get_paths(mwwid)
+        else:
+            disk_abs = disk = (self.disk.split("/dev/"))[1]
+            disk_nodes.append(disk)
+
+        if process.system("ls /dev/disk/by-id -l| grep -i %s" % disk_abs,
                           ignore_status=True, shell=True, sudo=True) != 0:
-            msg.append("Given disk %s is not in /dev/disk/by-path" % disk)
+            msg.append("Given disk %s is not in /dev/disk/by-id" % disk_abs)
+        for disk_node in disk_nodes:
+            if process.system("ls /dev/disk/by-path -l| grep -i %s" % disk_node,
+                              ignore_status=True, shell=True, sudo=True) != 0:
+                msg.append("Given disk %s is not in /dev/disk/by-path" % disk_node)
 
         # Verify disk listed in all tools
-        cmd_list = ["fdisk -l ", "parted -l", "lsblk ",
-                    "lshw -c disk "]
+        if mpath:
+            cmd_list = ["fdisk -l ", "lsblk "]
+        else:
+            cmd_list = ["fdisk -l ", "parted -l", "lsblk ",
+                        "lshw -c disk "]
```

Comment:
operate over disk_node, why mpath again?
we can still use the cmd_list as it is. not require to alter